### PR TITLE
New package: mlpack-3.2.2

### PR DESCRIPTION
**Pull request by toluschr**
If #20153 gets merged, the `makedepends` variable needs to be changed.

### CODE REVIEW

**Comment by toluschr** [comment]
Since I already got a review on #20153, I removed the extra dependency from this in advance.

**Comment by toluschr** [comment]
The build is failing, because #20153 didn't get merged yet. Nothing to worry about!

File: srcpkgs/mlpack/template [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+makedepends="armadillo-devel boost-devel"
+depends="armadillo boost lib${pkgname}>=${pkgname}-${version}"
+short_desc="Scalable C++ machine learning library"
+maintainer="toluschr <toluschr@protonmail.com>"
+license="BSD-3-Clause"
+homepage="https://mlpack.org/"
+distfiles="https://github.com/mlpack/mlpack/archive/${version}.tar.gz"
+checksum=ce0d950048edbffec9c45781245b1af402a7a30529bd2596fdc35ca9289cf02b
+
+nocross="usr/include/supermatrix.h: No such file or directory"
```
Comment (Chocimier) [inline srcpkgs/mlpack/template:16]: looks solvable with some CFLAGS, please remove for now

File: srcpkgs/mlpack/template [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Template file for 'mlpack'
+pkgname=mlpack
+version=3.2.2
+revision=1
+build_style=cmake
+hostmakedepends="txt2man"
+makedepends="armadillo-devel boost-devel"
+depends="armadillo boost lib${pkgname}>=${pkgname}-${version}"
```
Comment (Chocimier) [inline srcpkgs/mlpack/template:8]: it is autodetected: `SONAME: libmlpack.so.3 <-> libmlpack-3.2.2_1`

File: srcpkgs/mlpack/template [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	short_desc+=" - shared libraries"
+
+	pkg_install() {
+		vmove "usr/lib/*.so*"
+	}
+}
+
+libmlpack-devel_package() {
+	short_desc+=" - development files"
+	depends="${makedepends} lib${sourcepkg}>=${pkgname}-${version}"
```
Comment (Chocimier) [inline srcpkgs/mlpack/template:32]: did you mean libmlpack>=${version}_${revision} ?

File: srcpkgs/mlpack/template (hunk 1 above)
Comment (Chocimier) [reply srcpkgs/mlpack/template]: Should work with fixed armadillo, please rebase.

File: srcpkgs/mlpack/template [hunk 4]
```diff
@@ -0,0 +1,36 @@
+# Template file for 'mlpack'
+pkgname=mlpack
+version=3.2.2
+revision=1
+build_style=cmake
+hostmakedepends="txt2man"
+makedepends="armadillo-devel boost-devel"
+depends="armadillo boost"
```
Comment (Chocimier) [inline srcpkgs/mlpack/template:8]: armadillo and needed boost parts are autodetected too.

File: srcpkgs/mlpack/template [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+	short_desc+=" - shared libraries"
+
+	pkg_install() {
+		vmove "usr/lib/*.so*"
+	}
+}
+
+libmlpack-devel_package() {
+	short_desc+=" - development files"
+	depends="${makedepends} libmlpack>=${pkgname}_${version}"
```
Comment (Chocimier) [inline srcpkgs/mlpack/template:30]: should be rather `libmlpack>=${version}_${revision}`

Comment (toluschr) [reply srcpkgs/mlpack/template:30]: oh I completely messed up

**Comment by ericonr** [comment]
Ping @Chocimier

**Comment by Chocimier** [comment]
@toluschr 

>How well do it works for you?
It needs armadillo build with lapack for many operations.
Also would be great if you pack ensmallen as separate package.

Also 3.4.2 is out.

**Comment by toluschr** [comment]
@Chocimier I'm going to close the PR, since I'm not interested in this package anymore.

**Pull request closed without merging**